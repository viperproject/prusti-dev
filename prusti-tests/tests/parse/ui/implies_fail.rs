// compile-flags: -Pprint_desugared_specs=true -Pprint_typeckd_specs=true -Pno_verify=true -Phide_uuids=true
// normalize-stdout-test: "[a-z0-9]{32}" -> "$(NUM_UUID)"
// normalize-stdout-test: "[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}" -> "$(UUID)"
// normalize-stdout-test: "\[[a-z0-9]{4}\]::" -> "[$(CRATE_ID)]::"

use prusti_contracts::*;

#[requires(true ==>)]
fn test1() {}

#[requires(==> true)]
fn test2() {}

#[requires((true ==> true) || true)]
fn test3() {}

fn main() {}
