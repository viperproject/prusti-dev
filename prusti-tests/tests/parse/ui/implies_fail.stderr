error: expected expression
 --> $DIR/implies_fail.rs:8:17
  |
8 | #[requires(true ==>)]
  |                 ^^^

error: expected expression
  --> $DIR/implies_fail.rs:11:12
   |
11 | #[requires(==> true)]
   |            ^^

error: `==>` cannot be part of Rust expression
  --> $DIR/implies_fail.rs:14:18
   |
14 | #[requires((true ==> true) || true)]
   |                  ^^^

error: aborting due to 3 previous errors

